runtime/hashing/common/src/aws/smithy/kotlin/runtime/hashing/HashingSink.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.hashing
+
+import aws.smithy.kotlin.runtime.io.SdkSink
+import aws.smithy.kotlin.runtime.io.internal.SdkSinkObserver
+
+/**
+ * A sink which hashes data as it is being written to
+ * @param hash The [HashFunction] to hash data with
+ * @param sink the [SdkSink] to hash
+ */
+public class HashingSink(public val hash: HashFunction, sink: SdkSink) : HashFunction by hash, SdkSinkObserver(sink) {
```

Comment:
I'd probably make these `InternalApi`

Reply:
If anything I would create a `CrcSource/Sink` instead of trying to force them through `HashingSource/Sink`. It's going to look nearly identical but it won't be awkwardly special casing the generic hashing source/sink